cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1617,7 +1635,11 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 		return err
 	}
 
-	typs := []snap.Type{snap.TypeBase, snap.TypeGadget, snap.TypeKernel}
+	// order in the list must not change as it determines the mount order
+	typs := []snap.Type{snap.TypeGadget, snap.TypeKernel}
+	if !isClassic {
+		typs = append([]snap.Type{snap.TypeBase}, typs...)
+	}
```

Comment:
This is a bit confusing, because AFAIK we don't have gadget and kernel snaps on classic, do we? So, I would expect them also to be under the `!isClassic` condition.

Reply:
We will have them 🙂